### PR TITLE
gateways/repodata: remove load/save/read_mod_and_etag

### PR DESCRIPTION
**Pull request by dholth**
### Description

Remove deprecated functions, update tests.

#12681

### CODE REVIEW

File: tests/gateways/test_repodata_gateway.py [hunk 1]
```diff
@@ -175,13 +175,18 @@ def test_coverage_repodata_state(tmp_path):
...
+        "garbage.json",
     )
-    state.cache_path_state.write_text("not json")
-    assert dict(state.load()) == {}
+    state.load_state().cache_path_state.write_text("{}")
+
+    assert state.load_state().data == {}
+
+    # assert writing something loads same
+    state.cache_path_state.write_text('{"jim":"bob"}')
```
Comment (dholth) [inline tests/gateways/test_repodata_gateway.py:187]: This part fails so far. Maybe the mtime is wrong on one of the files or we got a filename wrong.

Check coverage.

File: conda/core/subdir_data.py [hunk 2]
```diff
@@ -576,34 +529,4 @@ def make_feature_record(feature_name):
         track_features=(feature_name,),
         build_number=0,
         fn=pkg_name,
-    )
-
-
-@deprecated(
-    "23.9",
-    "24.3",
```
Comment (dholth) [inline conda/core/subdir_data.py:584]: Whoops, this one is supposed to stick around longer.

File: conda/common/toposort.py [hunk 3]
```diff
@@ -81,7 +82,7 @@ def _safe_toposort(data):
             log.debug(err.args[0])
 
             if not data:
-                return
+                return  # pragma: nocover
```
Comment (dholth) [inline conda/common/toposort.py:85]: I think this is unreachable. Mark as # nocover to clean up the coverage report.

File: conda/core/subdir_data.py [hunk 4]
```diff
@@ -533,36 +518,6 @@ def _process_raw_repodata(self, repodata: dict, state: RepodataState | None = No
...
-            # ValueError: cannot mmap an empty file
-            return {}
-        except OSError as e:  # pragma: no cover
-            # OSError: [Errno 19] No such device
-            if e.errno == ENODEV:
-                return {}
-            raise
-
-
-def get_cache_control_max_age(cache_control_value: str):
```
Comment (dholth) [inline conda/core/subdir_data.py:561]: imported from other file; not really gone.

File: conda/common/toposort.py [hunk 5]
```diff
@@ -67,6 +67,7 @@ def _safe_toposort(data):
     dependencies, each subsequent set consists of items that depend upon
     items in the preceding sets.
     """
+
```
Comment (kenodegard) [inline conda/common/toposort.py:70]: making pre-commit happy

```suggestion
```

File: conda/core/subdir_data.py [hunk 6]
```diff
@@ -24,6 +21,9 @@
 except ImportError:  # pragma: no cover
     from .._vendor.boltons.setutils import IndexedSet
 
+from conda.gateways.repodata import (
+    get_cache_control_max_age,  # NOQA backwards compatibility alias
+)
```
Comment (kenodegard) [inline conda/core/subdir_data.py:26]: making pre-commit happy

```suggestion
from conda.gateways.repodata import (  # noqa: F401
    get_cache_control_max_age,  # backwards compatibility alias
)
```

File: conda/core/subdir_data.py [hunk 7]
```diff
@@ -7,12 +7,9 @@
 import pickle
 import re
```
Comment (kenodegard) [inline conda/core/subdir_data.py:8]: making pre-commit happy

```suggestion
```

**Comment by kenodegard** [comment]
pre-commit.ci autofix

**Review by kenodegard — APPROVED** [review]

**Pull request merged**